agents1/GenericAgent.py

```diff
@@ -325,47 +340,65 @@ def initialize_state(self, state):
                 "drop_off": state[block_name]['location']
             }
 
-            block_name = f"Collect_Block_{i+1}"
+            block_name = f"Collect_Block_{i + 1}"
 
     def phase_action(self, state):
+        msg = None
+        res = None
+
         if Phase.PLAN_PATH_TO_CLOSED_DOOR == self._phase:
-            return self.plan_path_to_closed_door(state, Phase.FOLLOW_PATH_TO_CLOSED_DOOR, Phase.PLAN_PATH_TO_OPEN_DOOR)
+            res = self.plan_path_to_closed_door(state, Phase.FOLLOW_PATH_TO_CLOSED_DOOR, Phase.PLAN_PATH_TO_OPEN_DOOR)
+            msg = self._mb.create_message(mt.MOVE_TO_ROOM, room_name=self._door['room_name'])
+
+        elif Phase.FOLLOW_PATH_TO_CLOSED_DOOR == self._phase:
+            res = self.follow_path(state, Phase.OPEN_DOOR)
 
-        if Phase.FOLLOW_PATH_TO_CLOSED_DOOR == self._phase:
-            return self.follow_path(state, Phase.OPEN_DOOR)
+        elif Phase.OPEN_DOOR == self._phase:
+            res = self.open_door(Phase.PLAN_ROOM_SEARCH)
+            msg = self._mb.create_message(mt.OPEN_DOOR, room_name=self._door['room_name'])
 
-        if Phase.OPEN_DOOR == self._phase:
-            return self.open_door(Phase.PLAN_ROOM_SEARCH)
+        elif Phase.PLAN_PATH_TO_OPEN_DOOR == self._phase:
+            res = self.plan_path_to_open_door(state, Phase.FOLLOW_PATH_TO_OPEN_DOOR, Phase.FOLLOW_PATH_TO_CLOSED_DOOR)
+            msg = self._mb.create_message(mt.MOVE_TO_ROOM, room_name=self._door['room_name'])
 
-        if Phase.PLAN_PATH_TO_OPEN_DOOR == self._phase:
-            return self.plan_path_to_open_door(state, Phase.FOLLOW_PATH_TO_OPEN_DOOR, Phase.FOLLOW_PATH_TO_CLOSED_DOOR)
+        elif Phase.FOLLOW_PATH_TO_OPEN_DOOR == self._phase:
+            res = self.follow_path(state, Phase.PLAN_ROOM_SEARCH)
 
-        if Phase.FOLLOW_PATH_TO_OPEN_DOOR == self._phase:
-            return self.follow_path(state, Phase.PLAN_ROOM_SEARCH)
+        elif Phase.PLAN_ROOM_SEARCH == self._phase:
+            res = self.plan_room_search(state, Phase.SEARCH_ROOM)
+            msg = self._mb.create_message(mt.SEARCHING_ROOM, room_name=self._door['room_name'])
 
-        if Phase.PLAN_ROOM_SEARCH == self._phase:
-            return self.plan_room_search(state, Phase.SEARCH_ROOM)
+        elif Phase.SEARCH_ROOM == self._phase:
+            res = self.search_room(state, Phase.PLAN_PATH_TO_BLOCK)
 
-        if Phase.SEARCH_ROOM == self._phase:
-            return self.search_room(state, Phase.PLAN_PATH_TO_BLOCK, Phase.PLAN_PATH_TO_CLOSED_DOOR)
+        elif Phase.PLAN_PATH_TO_BLOCK == self._phase:
+            res = self.plan_path(self._goal_blocks[self._searching_for]["location"], Phase.FOLLOW_PATH_TO_BLOCK)
+            msg = self._mb.create_message(mt.PICK_UP_BLOCK,
```

Comment:
we should set the message in the `GRAB_BLOCK` section, since we are not sure if the block is there yet